packages/vite/src/node/plugins/css.ts

```diff
@@ -732,6 +732,7 @@ function createCSSResolvers(config: ResolvedConfig): CSSAtImportResolvers {
           mainFields: ['style'],
           tryIndex: false,
           preferRelative: true,
+          skipExports: true,
```

Comment:
I don't think we can skip exports here. Exports should work when resolving in CSS as it's a convention, and it has higher precedence over `mainFields`. I'd argue this PR creates a wrong resolution instead.

Reply:
If I understand correctly, Vite does not follow any convention on styles and exports at the moment. The resolution of style entrypoints from the export map is blocked by #7817, which is under discussion. This PR fixes the `mainFields` styles resolution behavior, since it is currently broken in these cases. I would be happy to see #7817 merged, but if it is out of purpose for Vite, I think that the `mainFields` resolution should work at least.

Reply:
Vite does follow partial convention for styles, mainly the `sass`, `less`, `style` `mainFields` config. I still don't quite see what's wrong here, based on the `exports` spec, if `exports` field is present, `mainFields` is ignored completely. So if the package you're using has an `exports` field, it will resolve from `exports` only. We would be breaking the spec if we go with this PR.

https://nodejs.org/api/esm.html#resolver-algorithm-specification
> 5. If pjson is not null and pjson.exports is not null or undefined, then
>    1. Return the result of PACKAGE_EXPORTS_RESOLVE(packageURL, packageSubpath, pjson.exports, defaultConditions).
> 6. Otherwise, if packageSubpath is equal to ".", then
If pjson.main is a string, then
>    1. Return the URL resolution of main in packageURL.

Reply:
In my projects, prior to export maps, we were able to use the same package entrypoint for both CSS and JS with a top-level `main` and `style` fields. The constraint of Node's resolver algorithm that Vite is obeying here makes the pattern of having one entrypoint for both JS and style impossible when ESM exports are defined. Vite's current CSS resolver will always prefer the JavaScript `import` or `require` conditions from the `exports` map and ignore top level `style` fields. So to be consistent with Node's resolver spec, we need #7817.

However I would also argue that this PR isn't breaking any spec - the resources in question here are not ES Modules, they are stylesheets importing other stylesheets. When I write `@use` in Sass, I expect that my SCSS will resolve in accordance with Sass's own module resolution rules. The closest thing that Sass has decided there is some general consensus to use Node's own import algorithm https://github.com/sass/sass/issues/2739#issuecomment-839307125. In the absence of any real convention, I think this PR or #7187 are the only community-established ways forward.